wolfcrypt/src/sp_armthumb.c

```diff
@@ -291,28 +291,8 @@ SP_NOINLINE static void sp_2048_mul_8(sp_digit* r, const sp_digit* a,
         "\n"
     "L_sp_2048_mul_8_mul_%=: \n\t"
         "# Multiply Start\n\t"
-        "ldr	r6, [%[a]]\n\t"
-        "ldr	r7, [%[b]]\n\t"
-#ifdef __clang__
-        "lsls	r6, r6, #16\n\t"
-#else
-        "lsl	r6, r6, #16\n\t"
-#endif
-#ifdef __clang__
-        "lsls	r7, r7, #16\n\t"
-#else
-        "lsl	r7, r7, #16\n\t"
-#endif
-#ifdef __clang__
-        "lsrs	r6, r6, #16\n\t"
-#else
-        "lsr	r6, r6, #16\n\t"
-#endif
-#ifdef __clang__
-        "lsrs	r7, r7, #16\n\t"
-#else
-        "lsr	r7, r7, #16\n\t"
-#endif
+        "ldrh	r6, [%[a]]\n\t"
```

Comment:
Does this file need to be wrapped in `#ifndef WOLFSSL_KEIL` to prevent both .c files from being compiled? Specifically with wildcard *.c includes like Eclipse this could be an issue.

Reply:
Updated sp_armthumb.c as requested.